File: Class 2/src/mainwindow.cpp
#include "mainwindow.h"
#include "ui_mainwindow.h"
#include "latexwidget.h"
#include "functionselector.h"
#include "graphwidget.h"

#include <QMetaObject>

MainWindow::MainWindow(QWidget *parent)
    : QMainWindow(parent),
      functionSelector_(new FunctionSelector(this)),
      graphWidget_(new GraphWidget(this)),
      ui(new Ui::MainWindow)
{
    ui->setupUi(this);

    // Add tabs
    ui->tabWidget->addTab(new LatexWidget(this), "LaTeX Viewer");
    ui->tabWidget->addTab(functionSelector_, "Function Selector");
    ui->tabWidget->addTab(graphWidget_, "Graph Viewer");

    connect(functionSelector_, &FunctionSelector::functionSelected, graphWidget_, &GraphWidget::setPlotDefinition);
    connect(functionSelector_, &FunctionSelector::functionSelected, this, [this]() {
        ui->tabWidget->setCurrentWidget(graphWidget_);
    });

    QMetaObject::invokeMethod(functionSelector_, "emitSelection", Qt::QueuedConnection);
}

MainWindow::~MainWindow() {
    delete ui;
}

File: Class 2 - versao 2/src/class2_v2_app/main_window.py
from __future__ import annotations

from PyQt6.QtWidgets import QMainWindow, QTabWidget

from .function_selector import FunctionSelector
from .graph_widget import GraphWidget
from .latex_widget import LatexWidget


class MainWindow(QMainWindow):
    def __init__(self) -> None:
        super().__init__()
        self.setWindowTitle("Class 2 - versao 2")
        self.resize(960, 680)

        self.tabs = QTabWidget(self)
        self.setCentralWidget(self.tabs)

        self.latex_widget = LatexWidget(self)
        self.function_selector = FunctionSelector(self)
        self.graph_widget = GraphWidget(self)

        self.tabs.addTab(self.latex_widget, "LaTeX Viewer")
        self.tabs.addTab(self.function_selector, "Function Selector")
        self.tabs.addTab(self.graph_widget, "Graph Viewer")

        self.function_selector.function_selected.connect(self.graph_widget.set_plot_definition)
        self.function_selector.function_selected.connect(self._show_graph_tab)
        self.function_selector.emit_selection()

    def _show_graph_tab(self, *_args) -> None:
        self.tabs.setCurrentWidget(self.graph_widget)
